### PR TITLE
Demangling C++ function names in stack traces

### PR DESCRIPTION
**Pull request by ElNounch**
Using abi::__cxa_demangle

### CODE REVIEW

File: src/OSSupport/StackTrace.cpp [hunk 1]
```diff
@@ -34,11 +36,35 @@ void PrintStackTrace(void)
...
+
 			auto numItems = backtrace(stackTrace, ARRAYCOUNT(stackTrace));
-			backtrace_symbols_fd(stackTrace, numItems, STDERR_FILENO);
+			char** buffer = backtrace_symbols(stackTrace, numItems);
+
+			for (auto frameId = (numItems - 1); frameId > 0; frameId--)
+			{
+				bool handled = false;
+				std::cmatch m;
+				if (std::regex_match (buffer[frameId], m, lineSplitter))
```
Comment (madmaxoft) [inline src/OSSupport/StackTrace.cpp]: Do we need to match things with a regex, can't we just send everything for demangling?

I just fear that once a crash is hit, nothing can be considered stable enough, especially memory allocation.

Comment (ElNounch) [inline src/OSSupport/StackTrace.cpp]: Shouldn't use backtrace or Stackwalker then, but a wrapper shell script :
```shell
#!/bin/sh
ulimit -c unlimited
$@
if [ -e core ]; then
    gdb $1 ./core -ex backtrace -ex quit
    echo Compressing and renaming 'core' file...
    gzip --best core;
    mv core.gz core-$(stat -c '%Y' core.gz).gz;
fi
```

Comment (madmaxoft) [inline src/OSSupport/StackTrace.cpp]: Okay, still, there's my first question: Why regex at all, why not send everything in?

**Comment by ElNounch** [comment]
Need to isolate the mangled name part, yes.

File: src/OSSupport/StackTrace.cpp (hunk 1 above)
Comment (ElNounch) [inline src/OSSupport/StackTrace.cpp]: Need to isolate the mangled name part, yes.

**Comment by sphinxc0re** [comment]
This PR and #3344 look promising. Any updates? @ElNounch

**Pull request closed without merging**